### ISSUE TITLE
Bug: [0.9.0] No SIGTERM handling during IBD

### ISSUE THREAD

**Issue by erikarvstedt** [romanz/electrs#532]
Electrs 0.9.0 doesn't respond to `SIGTERM` during IBD and keeps running. After IBD, `SIGTERM` works as exepcted.

Log output:
```
Config { network: Bitcoin, db_path: "/var/lib/electrs/bitcoin", daemon_dir: "/var/lib/bitcoind", daemon_auth: UserPass("public", "<sensitive>"), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 127.0.0.1:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: Some(200), auto_reindex: true, ignore_mempool: false, sync_once: false, server_banner: "Welcome to electrs 0.9.0 (Electrum Rust Server)!", args: [] }
....
[2021-10-05T10:59:02.077Z INFO  electrs::daemon] waiting for 364000 blocks to download (IBD)
[2021-10-05T10:59:03.077Z INFO  electrs::daemon] waiting for 388000 blocks to download (IBD)
[2021-10-05T10:59:03.390Z INFO  electrs::signals] notified via SIG15
[2021-10-05T10:59:04.081Z INFO  electrs::daemon] waiting for 414000 blocks to download (IBD)
[2021-10-05T10:59:05.082Z INFO  electrs::daemon] waiting for 440000 blocks to download (IBD)
....
```

**Comment by Kixunil** [comment]
Did you wait long enough? The signal is processed asynchronously so it can take time for it to have effect. Went to check if #498 handles both SIGINT and SIGTERM and it seems it should but just in case have you tried SIGINT?

**Comment by erikarvstedt** [comment]
[This script](https://gist.github.com/4880be392304395dcfec31bf02c307be) reproduces the bug.

```
[2021-10-05T14:56:29.904Z INFO  electrs::signals] notified via SIG15
[2021-10-05T14:56:29.909Z INFO  electrs::daemon] waiting for 0 blocks to download (IBD)
[2021-10-05T14:56:30.911Z INFO  electrs::daemon] waiting for 0 blocks to download (IBD)
...
```
electrs is then `sigkill`ed after 2 minutes.

**Comment by romanz** [comment]
Good catch, thanks!

**Comment by romanz** [comment]
BTW, it is still not possible to gracefully exit during DB full compaction (which takes place after the initial sync is over) :(

**Comment by Kixunil** [comment]
How long does compaction take on common hardware?

**Comment by romanz** [comment]
~40 minutes on an HDD (most of the time blocked on I/O):
https://github.com/romanz/electrs/blob/45a91f9b1b99034bc803204597b766c2083fb48a/doc/usage.md#L300-L307

**Comment by Kixunil** [comment]
That's really quite long to wait for shutdown, would be nice to enable stopping in the middle. If I understand correctly `self.db.compact_range_cf(cf, None::<&[u8]>, None::<&[u8]>);` could take `Some(chunk_start), Some(chunk_end)`. We could split it up into e.g. 100 chunks (maybe configurable) and check exit flag between each. Would it add too big overhead?

**Comment by romanz** [comment]
> That's really quite long to wait for shutdown, would be nice to enable stopping in the middle. 

I'll try running initial sync with auto-compactions... if it won't increase the sync time signficantly, we can switch to it - simplifying DB-related code.
